config/config_test.go

```diff
@@ -75,72 +100,96 @@ func TestNewConfigDefaultConfig(t *testing.T) {
-	err := NewConfigFromEnv()
+	cfg, err := NewConfigFromEnv()
 	assert.NoError(t, err)
+	assert.NotNil(t, cfg)
 
-	assert.Equalf(t, DefaultRunMode, FTWConfig.RunMode,
-		"unexpected default value '%s' for run mode", FTWConfig.RunMode)
-	assert.Equalf(t, DefaultLogMarkerHeaderName, FTWConfig.LogMarkerHeaderName,
-		"unexpected default value '%s' for logmarkerheadername", FTWConfig.LogMarkerHeaderName)
+	assert.Equalf(t, DefaultRunMode, cfg.RunMode,
+		"unexpected default value '%s' for run mode", cfg.RunMode)
+	assert.Equalf(t, DefaultLogMarkerHeaderName, cfg.LogMarkerHeaderName,
+		"unexpected default value '%s' for logmarkerheadername", cfg.LogMarkerHeaderName)
 
 }
 
 func TestNewConfigFromFileHasDefaults(t *testing.T) {
 	filename, _ := utils.CreateTempFileWithContent(yamlConfig, "test-*.yaml")
 	defer os.Remove(filename)
 
-	err := NewConfigFromFile(filename)
+	cfg, err := NewConfigFromFile(filename)
 	assert.NoError(t, err)
-
-	assert.Equalf(t, DefaultRunMode, FTWConfig.RunMode,
-		"unexpected default value '%s' for run mode", FTWConfig.RunMode)
-	assert.Equalf(t, DefaultLogMarkerHeaderName, FTWConfig.LogMarkerHeaderName,
-		"unexpected default value '%s' for logmarkerheadername", FTWConfig.LogMarkerHeaderName)
+	assert.NotNil(t, cfg)
+	assert.Equalf(t, DefaultRunMode, cfg.RunMode,
+		"unexpected default value '%s' for run mode", cfg.RunMode)
+	assert.Equalf(t, DefaultLogMarkerHeaderName, cfg.LogMarkerHeaderName,
+		"unexpected default value '%s' for logmarkerheadername", cfg.LogMarkerHeaderName)
 }
 
 func TestNewConfigFromStringHasDefaults(t *testing.T) {
-	err := NewConfigFromString("")
+	cfg, err := NewConfigFromString("")
 	assert.NoError(t, err)
-
-	assert.Equalf(t, DefaultRunMode, FTWConfig.RunMode,
-		"unexpected default value '%s' for run mode", FTWConfig.RunMode)
-	assert.Equalf(t, DefaultLogMarkerHeaderName, FTWConfig.LogMarkerHeaderName,
-		"unexpected default value '%s' for logmarkerheadername", FTWConfig.LogMarkerHeaderName)
+	assert.NotNil(t, cfg)
+	assert.Equalf(t, DefaultRunMode, cfg.RunMode,
+		"unexpected default value '%s' for run mode", cfg.RunMode)
+	assert.Equalf(t, DefaultLogMarkerHeaderName, cfg.LogMarkerHeaderName,
+		"unexpected default value '%s' for logmarkerheadername", cfg.LogMarkerHeaderName)
 }
 
 func TestNewConfigFromFileRunMode(t *testing.T) {
 	filename, _ := utils.CreateTempFileWithContent(yamlCloudConfig, "test-*.yaml")
 	defer os.Remove(filename)
 
-	err := NewConfigFromFile(filename)
+	cfg, err := NewConfigFromFile(filename)
 	assert.NoError(t, err)
+	assert.NotNil(t, cfg)
+	assert.Equalf(t, CloudRunMode, cfg.RunMode,
+		"unexpected value '%s' for run mode, expected '%s;", cfg.RunMode, CloudRunMode)
+}
 
-	assert.Equalf(t, CloudRunMode, FTWConfig.RunMode,
-		"unexpected value '%s' for run mode, expected '%s;", FTWConfig.RunMode, CloudRunMode)
+func TestNewDefaultConfigWithParams(t *testing.T) {
+	cfg := NewDefaultConfig()
+	assert.Equal(t, DefaultLogMarkerHeaderName, cfg.LogMarkerHeaderName)
```

Comment:
You have a separate test for the default configuration. I wouldn't repeat the assertions from that test here.

Reply:
Removed.